control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -394,6 +394,28 @@ impl ResourceSpecsLocked {
             .cloned()
             .collect()
     }
+
+    /// Get a list of protected NexusSpecs's which are associated with the given volume `id`
+    /// and are currently in shutdown state.
+    pub(crate) async fn get_volume_shutdown_nexuses_guards(
+        &self,
+        id: &VolumeId,
+    ) -> Result<Vec<OperationGuardArc<NexusSpec>>, SvcError> {
+        let mut shutdown_nexuses = Vec::new();
+        for nexus in self.read().nexuses.values() {
+            let nexus_spec = nexus.lock().clone();
+            if nexus_spec.name == id.to_string() && nexus_spec.is_shutdown() {
+                shutdown_nexuses.push(nexus_spec);
+            }
+        }
+        let mut shutdown_nexuses_guards = Vec::with_capacity(shutdown_nexuses.len());
+        for nexus in shutdown_nexuses {
+            let nexus_guard = self.nexus(&nexus.uuid).await?;
```

Comment:
this will error out if we can't grab one of the nexuses, is this the right thing to do?
What about returning only the Resource<Nexus>, and then try to guard them one at a time, and if we fail to guard one of them skip and fail the shutdown call, but at least we've cleaned up some of them?